cli/cfncluster/cfncluster.py

```diff
@@ -403,6 +413,54 @@ def instances(args):
     for instance in instances:
         print('%s         %s' % (instance[0],instance[1]))
 
+def get_head_ip(outputs):
+    for output in outputs:
+        if output.key == "MasterPublicIP":
+            return output.value
+
+def get_head_user(parameters):
+    base_os = [i.value for i in parameters if i.key == "BaseOS"][0]
+    return username_map[base_os]
+
+def login(args):
+    stack = ('cfncluster-' + args.cluster_name)
+    config = cfnconfig.CfnClusterConfig(args)
+    cfnconn = boto.cloudformation.connect_to_region(config.region,aws_access_key_id=config.aws_access_key_id,
+                                                    aws_secret_access_key=config.aws_secret_access_key)
+    try:
+        status = cfnconn.describe_stacks(stack)[0].stack_status
+        if not ((status == 'CREATE_COMPLETE') or (status == 'UPDATE_COMPLETE')):
```

Comment:
How about using DescribeInstanceStatus or some other mechanism to just wait for the master node to come up before allowing logins instead of waiting for the entire cluster to be ready? It will be convenient to login to the master server and prepare my batch scripts/compile my code while my large compute ASG is spinning up.

Reply:
Just a thought,  using DescribeInstanceStatus is good idea, just want to make sure associate-address happened w.r.t eip

Reply:
I did not address this in the latest pr as I want to get the interface correct first.